servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/AbstractRestReferences.java

```diff
@@ -48,11 +54,75 @@
 import org.projectnessie.model.Operation.Put;
 import org.projectnessie.model.Reference;
 import org.projectnessie.model.ReferenceMetadata;
+import org.projectnessie.model.ReferencesResponse;
 import org.projectnessie.model.Tag;
 import org.projectnessie.model.Validation;
 
 /** See {@link AbstractTestRest} for details about and reason for the inheritance model. */
 public abstract class AbstractRestReferences extends AbstractRestMisc {
+  @NessieApiVersions(versions = NessieApiVersion.V2)
+  @ParameterizedTest
+  @ValueSource(ints = {0, 20, 22})
+  public void referencesPaging(int numRefs) throws BaseNessieClientServerException {
+    try {
+      getApi().getAllReferences().pageToken("Zm9v").maxRecords(20).get();
+    } catch (NessieBadRequestException e) {
+      if (!e.getMessage().contains("Paging not supported")) {
```

Comment:
Is the `@NessieApiVersions` annotation not sufficient to ensure this test runs only when paging is supported?

Reply:
Nope. Current backends do not support paging :(